File: app/src/main/java/br/com/vicentec12/mygames/ui/game/GameAdapter.java
package br.com.vicentec12.mygames.ui.game;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;
import androidx.recyclerview.widget.ListAdapter;

import br.com.vicentec12.mygames.data.model.Game;
import br.com.vicentec12.mygames.databinding.ItemGameBinding;
import br.com.vicentec12.mygames.interfaces.OnItemClickListener;
import br.com.vicentec12.mygames.interfaces.OnItemLongClickListener;

public class GameAdapter extends ListAdapter<Game, GameHolder> {

    private final GameViewModel mViewModel;

    private OnItemClickListener mOnItemClickListener;
    private OnItemLongClickListener mOnItemLongClickListener;

    public GameAdapter(GameViewModel mViewModel) {
        super(Game.getDiffItemCallback());
        this.mViewModel = mViewModel;
    }

    @NonNull
    @Override
    public GameHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemGameBinding mBinding = ItemGameBinding.inflate(LayoutInflater.from(parent.getContext()),
                parent, false);
        return new GameHolder(mBinding, mOnItemClickListener, mOnItemLongClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull GameHolder holder, int position) {
        holder.bindGame(getItem(position), mViewModel);
    }

    // Responsável por resolver glichs quando notifyDataSetChange era chamado
    @Override
    public long getItemId(int position) {
        Game game = getItem(position);
        return game.getId();
    }

    public void setOnItemClickListener(OnItemClickListener mOnItemClickListener) {
        this.mOnItemClickListener = mOnItemClickListener;
    }

    public void setOnItemLongClickListener(OnItemLongClickListener onItemLongClickListener) {
        this.mOnItemLongClickListener = onItemLongClickListener;
    }

}

File: app/src/main/java/br/com/vicentec12/mygames/ui/game/GameAdapter.kt
package br.com.vicentec12.mygames.ui.game

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import br.com.vicentec12.mygames.data.model.Game
import br.com.vicentec12.mygames.databinding.ItemGameBinding
import br.com.vicentec12.mygames.interfaces.OnItemClickListener
import br.com.vicentec12.mygames.interfaces.OnItemLongClickListener

class GameAdapter(
        private val mViewModel: GameViewModel,
        private val mOnItemClickListener: OnItemClickListener?,
        private val mOnItemLongClickListener: OnItemLongClickListener?
) : ListAdapter<Game, GameHolder>(Game.DIFF_UTIL_CALLBACK) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = GameHolder(
            ItemGameBinding.inflate(LayoutInflater.from(parent.context), parent, false),
            mOnItemClickListener,
            mOnItemLongClickListener
    )

    override fun onBindViewHolder(holder: GameHolder, position: Int) {
        holder.bind(getItem(position), mViewModel)
    }

    override fun getItemId(position: Int) = getItem(position).id

}